client/src/components/Search/TagsSearch/TagsSearch.js

```diff
@@ -40,30 +40,59 @@ class TagsFilter extends React.Component {
     componentDidUpdate(prevProps, prevState, snapshot) {
         if ((!prevProps.selectedTags || !prevProps.selectedTags.length) &&
             this.props.selectedTags && this.props.selectedTags.length) {
-                this.setState({
-                    selectedTags: (this.state.selectedTags || []).concat(this.props.selectedTags)
-                        .map(tag => {
-                            tag.selected = true;
-                            return tag;
-                        })
-                })
+            this.setState({
+                selectedTags: (this.state.selectedTags || []).concat(this.props.selectedTags)
+                    .map(tag => {
+                        tag.selected = true;
+                        return tag;
+                    })
+            })
         }
     }
 
     render() {
-        const { searchQuery, selectedTags } = this.state;
-        const { searchResults, variant } = this.props;
+        const { searchQuery, selectedTags, createdTags, createdTagsKey } = this.state;
+        const { searchResults, variant, className, allowTagCreation } = this.props;
         const visibleSearchResults = searchResults.filter(result => {
             return !selectedTags.find(tag => tag.id === result.id);
         });
 
```

Comment:
This section below adds the ability to create a new Tag/Material from the `TagsSearch` component.